### PR TITLE
Dm obj alloc

### PR DESCRIPTION
**Pull request by thomas-robinson**
**Description**
Adds has_* functions for daig object and yaml objects

**How Has This Been Tested?**
Tested with make check using mpiifort and mpiicc 

**Checklist:**
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] Any dependent changes have been merged and published in downstream modules
- [x] New check tests, if applicable, are included
- [x] `make distcheck` passes

### CODE REVIEW

**Comment by thomas-robinson** [comment]
@rem1776 why isn't the CI set up to run on this PR?

File: diag_manager/fms_diag_object.F90 [hunk 1]
```diff
@@ -67,8 +67,8 @@ module fms_diag_object_mod
      class(*), allocatable, private                   :: missing_value     !< The missing fill value
      class(*), allocatable, private                   :: data_RANGE        !< The range of the variable data
      type (diag_axis_type), allocatable, dimension(:) :: axis              !< The axis object
-!> \brief Extends the variable object to work with multiple types of data
-     class(*), allocatable :: vardata0
+! data buffers
+     class(*), allocatable :: vardata0
```
Comment (uramirez8707) [inline diag_manager/fms_diag_object.F90:71]: This is adding leading whitespace

**Review by uramirez8707 — CHANGES_REQUESTED** [review]

File: diag_manager/fms_diag_yaml.F90 [hunk 2]
```diff
@@ -979,7 +979,7 @@ end function has_file_duration
 !> @brief Checks if obj%file_duration_units is allocated
 !! @return true if obj%file_duration_units is allocated
 pure logical function has_file_duration_units (obj)
-  type(diagYamlfile_duration_type), intent(out) :: obj !< diagYamlfile_duration_type object to initialize
+  type(diagYamlFiles_type), intent(out) :: obj !< diagYamlFiles_type object to initialize
```
Comment (ngs333) [inline diag_manager/fms_diag_yaml.F90]: intent may be incorrect?

**Review by ngs333 — APPROVED** [review]
The has_ functions look OK. Given their existence and convention, some ofhter functions may eventually be modified. E.g. how should functions is_registered be coded?

File: diag_manager/fms_diag_object.F90 (hunk 1 above)
Comment (thomas-robinson) [reply diag_manager/fms_diag_object.F90]: This was just all wrong anyway.  The data buffers now all have their own comments.

**Comment by thomas-robinson** [comment]
> The has_ functions look OK. Given their existence and convention, some other functions may eventually be modified. E.g. how should function is_registered() be coded?

We can talk about this offline in a meeting I think.

**Pull request merged**